### ISSUE TITLE
1.4 doesn't allow binary operations on literals

### ISSUE THREAD

**Issue by PavelSarek** [influxdata/influxdb#9290]
### Bug report

__System info:__ Any 1.4.x or the last 1.5 nightly build

__Steps to reproduce:__

Execute query:
`select count * (1 / 1) from measurement`
or
`select * from measurement where 'a'='a'`

__Expected behavior:__ The same as in the previous versions, i.e. 1.3.x and older. The expression which can be reduced to a constant during AST processing is accepted and correctly evaluated. `'a'='a'` would be equivalent to `true`, `count * (1 / 1)` would be the same as `count * 1`

__Actual behavior:__ The expression is refused by the parser. With `ERR: cannot perform a binary expression on two literals`, or respectively `ERR: invalid expression: 'a' = 'a'`

__Additional info:__ 
Might be related to #9226 

Rule `binary_op` in InfluxQL reference .

__Use case:__ Some use-cases in Grafana using templating variables. 

The first query is derived from autoscaling implementation:
```
SELECT sum("count")/($__interval_ms/3600000) GROUP BY time($__interval)
```
the second is used to add some modes to Grafana dashboard by switching some queries on/off based on templating variables.

**Comment by prokod** [comment]
@rbetts @jsternberg Could you please have a look on that ? Thx

**Comment by jsternberg** [comment]
@rbetts this seems like a candidate for backporting.

**Comment by jasonbogdanski** [comment]
@PavelSarek Did you happen to find a workaround for auto scaling in Grafana? This is still an issue in Influxdb 1.4.3.

**Comment by PavelSarek** [comment]
@jasonbogdanski I'm afraid I have no workaround. We reverted to InfluxDB 1.3.7 for now.

**Comment by jasonbogdanski** [comment]
Thanks @PavelSarek, we were thinking we going to have to do the same thing.

**Comment by jsternberg** [comment]
This should be fixed in 1.4.3. @PavelSarek can you give me a reproducer?

I also just tested the example in the first post and it appears like this query returned zero results in 1.3.7. But, I'm not getting a parsing failure.

**Comment by jasonbogdanski** [comment]
@jsternberg I'm running a docker image of influxdb v1.4.3 from https://hub.docker.com/_/influxdb/.  I can reproduce the issues using that version with `select count / (1 / 1) from measurement`.  This issue was also posted again on https://github.com/influxdata/influxdb/issues/9273, which was tagged for a 1.6.0 release. However, I also seen it tagged on the v1.5.0 release.

**Comment by benjarrell** [comment]
@jsternberg I tried with 1.4.3 and a nightly (1.6.0~n201802060800) RPMs and still gives the same error using `SELECT non_negative_derivative(mean("bytes_sent"), 10s) * 8 / (10000 / 1000)  FROM "mesurement"`

**Comment by andyxning** [comment]
This is quit a non-backward compatible change. And it should be fixed ASAP.

**Comment by rbetts** [comment]
@andyxning we agree. This will be fixed and backported.

**Comment by jasonbogdanski** [comment]
@rbetts When do you except this will be fixed and backported?

**Comment by andyxning** [comment]
@rbetts Seems this is a regression issue. Any ETA about this and which release will have this fixed?

**Comment by jsternberg** [comment]
To clarify, there are two behaviors here that are both bugs. This first one:

```
select count * (1 / 1) from measurement
```

This was fixed and backported to 1.4.3.

This one:

```
select * from measurement where 'a'='a'
```

This did not produce an error in 1.3, but it also didn't work correctly to begin with. So the regression, which was it causing an error, was fixed and backported to 1.4.3. I've put up a few pull requests to fix the actual behavior so a true literal returns everything. This should be present in 1.5.0 and we will backport it to the 1.4.x line (so likely 1.4.4).

**Comment by andyxning** [comment]
Thanks for you explanation @jsternberg . Any ETA about the release of 1.4.4. I am willing to give it a try. :)

**Comment by rbetts** [comment]
@andyxning we are wrapping up a 1.5.0 (we built 1.5.0rc4 have been testing internally). This fix will make the 1.5.0 final which should be available on OSS this week if all goes well.

**Comment by andyxning** [comment]
@rbetts Good news. 👍
